### DISCUSSION TITLE
action可以自动更新吗

### DISCUSSION THREAD

**Discussion by wx-11** [dreamhunter2333/cloudflare_temp_email#526] [Q&A] [answered]
我是项目刚开始的时候部署的 手动部署我记得还挺麻烦
现在发现热度很高 更新了非常多东西 但是没有简单的更新步骤
看文档 是需要每一个版本差异都要一个个执行sql? 我就想action是不是会自动更新
不过看了一下好像是一次性部署 并不是自动更新
请问一下
- 更新是否需要一个个执行每一个版本的sql
- action,可以做到自动更新吗
- 不想一个个更新想重新部署最新版 可以保留之前的数据库邮箱吗

谢谢开发项目 不想打扰issue 所以来这里提问了 希望没有打扰

**Comment by dreamhunter2333** [comment, marked as answer]
1. 需要一个个执行每一个版本的sql, 数据库不会经常变更
2. action 目前没有做自动更新
3. 更新不会清空数据，绑定相同的数据库即可

**Comment by dreamhunter2333** [comment]
增加了 github 自动部署 main 分支的功能，有空可以试试
https://temp-mail-docs.awsl.uk/zh/guide/github-action.html

**Reply by wx-11** [reply]
> 增加了 github 自动部署 main 分支的功能，有空可以试试 https://temp-mail-docs.awsl.uk/zh/guide/github-action.html

哦哦 这个好像只是自动更新上游 数据库还是需要手动 我想的是 action会根据更新文件的.sql自动执行. 不然自动同步部署但是数据库不兼容其实手动会比较好.

不过sql也没怎么更新 应该够用了 感谢大佬

**Comment by wx-11** [comment]
好滴感谢!

关于3我的意思其实是 不一个个执行sql. 直接重新部署 但是用之前的数据库 
不过这样好像还是要 一个个更新数据库 不然可能不兼容 我一个个先更新一下 谢谢

**Reply by dreamhunter2333** [reply]
是的，我考虑后续增加个自动更新数据库 schema 的 action

**Reply by wx-11** [reply]
> 续增加个自动更新数据库 sche

那太棒了 如果是这样 action就是主力更新步骤了 因为自有源码 可以根据文件自动判断 这个思路挺好的 期待
可惜不会什么技术 不然也能帮上忙了 .. 大佬有空再弄 不着急
